abe/ma-abe_test.go

```diff
@@ -0,0 +1,172 @@
+package abe_test
+
+import (
+    "testing"
+    "github.com/fentec-project/gofe/abe"
+    "github.com/stretchr/testify/assert"
+)
+
+func TestMAABE(t *testing.T) {
+    // create new MAABE struct with Global Parameters
+    maabe := abe.NewMAABE()
+
+    // create three authorities, each with two attributes
+    attribs1 := []string{"auth1:at1", "auth1:at2"}
+    attribs2 := []string{"auth2:at1", "auth2:at2"}
+    attribs3 := []string{"auth3:at1", "auth3:at2"}
+    auth1, err:= maabe.NewMAABEAuth("auth1", attribs1)
+    if err != nil {
+        t.Fatalf("Failed generation authority %s: %v\n", "auth1", err)
+    }
+    auth2, err:= maabe.NewMAABEAuth("auth2", attribs2)
+    if err != nil {
+        t.Fatalf("Failed generation authority %s: %v\n", "auth2", err)
+    }
+    auth3, err:= maabe.NewMAABEAuth("auth3", attribs3)
+    if err != nil {
+        t.Fatalf("Failed generation authority %s: %v\n", "auth3", err)
+    }
+
+    // create a msp struct out of the boolean formula
+    msp, err := abe.BooleanToMSP("((auth1:at1 AND auth2:at1) OR (auth1:at2 AND auth2:at2)) OR (auth3:at1 AND auth3:at2)", false)
+    if err != nil {
+        t.Fatalf("Failed to generate the policy: %v\n", err)
+    }
+
+    // define the set of all public keys we use
+    pks := []*abe.MAABEPubKey{auth1.Pk, auth2.Pk, auth3.Pk}
+
```

Comment:
Maybe create a function `auth.PubKeys()` that returns all the public keys of the authority.